service/raft/raft_group0.cc

```diff
@@ -55,7 +56,15 @@ raft_server_for_group raft_group0::create_server_for_group(raft::group_id gid,
 
     _raft_gr.address_map().set(my_addr);
     auto state_machine = std::make_unique<group0_state_machine>(_mm, _qp.proxy());
-    auto rpc = std::make_unique<raft_rpc>(*state_machine, _ms, _raft_gr.address_map(), gid, my_addr.id);
+    auto rpc = std::make_unique<raft_rpc>(*state_machine, _ms, _raft_gr.address_map(), gid, my_addr.id,
+            [this] (gms::inet_address addr, bool added) {
+                auto id = _gossiper.get_direct_fd_pinger().allocate_id(addr);
```

Comment:
There can be two different hosts with the same IP in the system, for example when replacing a node with the same IP. I don't think it's currently handled, right?

Reply:
The GOSSIP_ECHO RPC contains a generation number which I think is supposed to solve the problem of replacing nodes, i.e. if the node which receives the RPC is a replacing node and the ping was intended for the replaced node, the node was supposed to somehow detect that it's not the desired destination because there would be a generation number mismatch.

From reading the code however (`gossiper::handle_echo_msg`) I don't think it works.

We discussed this with Gleb already, and we agreed that it's fine to leave this problem to a follow-up. One possible solution is to introduce a new verb for failure detector pings. The RPC would send the destination's raft server ID when calling the node. The node would receive the ID and compare with its own. It would only answer if they are the same. So if a replacing node receives an RPC which was intended for replaced node, it would receive a different server ID and decide not to answer.

Reply:
On RPC level, the generic solution is https://github.com/scylladb/scylla/issues/6403. Passing ids around is an alternative implementation, but it's better to have a central mechanism rather than have each verb implement it.

Even if we solve it at RPC level, the pinger's registry should also work with UUIDs instead of inet_address. 

We can do it as a follow up, but please leave a FIXME here.

Reply:
ok